### PR TITLE
Support applet install and uninstall

### LINKED ISSUE #577: Support dynamic loading and unloading of applets
This issue is the intersection of #56 and #38. This issue does not try to support multiple applets (or if so, with a maximum of one applet). This would be done in a separate issue.

### PR DESCRIPTION
**Pull request by ia0**
Fixes #577

### CODE REVIEW

File: crates/cli-tools/CHANGELOG.md [hunk 1]
```diff
@@ -8,9 +8,15 @@
...
+- Add `action::PlatformLock` for locking a platform protocol
+- Expose `action::Transfer` for transfers from host to device
+- Add `action::Wait::set_period()` and `impl Default for action::Wait`
+- Add `action::AppletExitStatus` to get the applet exit status
+- Add `action::Applet{Install,Uninstall}` for simple applet management
+- Add `action::PlatformApiVersion` to get a platform API version
 - Change the flags of `action::AppletRpc` to use `action::Wait`
 - Add `action::Wait` for commands returning an optional response
-- Increase the default connection timeout from 1 to 5 seconds
+- Change the default connection timeout from 1 seconds to infinite (0 seconds)
```
Comment (ia0-review) [inline crates/cli-tools/CHANGELOG.md:19]: typo

File: crates/cli-tools/CHANGELOG.md [hunk 2]
```diff
@@ -8,9 +8,15 @@
 
 ### Minor
 
+- Add `action::PlatformLock` for locking a platform protocol
+- Expose `action::Transfer` for transfers from host to device
+- Add `action::Wait::set_period()` and `impl Default for action::Wait`
```
Comment (ia0-review) [inline crates/cli-tools/CHANGELOG.md:13]: already contained below

**Review by ia0-review — CHANGES_REQUESTED** [review]
Only reviewed up to and including `crates/cli`. Will do another iteration after fixes.

**Review by ia0-review — APPROVED** [review]

**Comment by ia0** [comment]
There are 2 issues:
- Footprint logic must be updated to take into account that wasm applets are not embedded anymore.
- Seems like using unix socket on Github doesn't work (maybe use something else than `/tmp/wasefire`).

**Comment by ia0** [comment]
There's a few issues that might be worse fixing after this PR:
- `protocol-usb` is used with both `device` and `host`. The feature documentation in `Cargo.toml` should be fixed and a test with both features should be added to `test.sh`.
- `ensure_applet` assumes `thumbv7em-none-eabi` which is not always true. We use `i686-unknown-linux-gnu` in `scheduler` and `runner-host`. This should be cleaned up to use an appropriate library.
- https://github.com/a1ien/rusb/issues/211 may either be related to the above 2 points or something else (but it seems the runner image is the same as yesterday where the test was passing, so it must be a change in this PR and it must be specific to `runner-host` since `scheduler` is fine, so probably the first point rather than the second point)

**Review by ia0-review — APPROVED** [review]

**Comment by ia0** [comment]
Ok the issues have been understood: 
- #619 fixes the first issue 
- `cargo check` doesn't care what `libapplet.a` is as long as it exists, so we should just touch it
- the issue was that we compiled `rusb` in a device, this is fixed by #619 (initially fixed by #618 but the wrong fix)

**Comment by ia0** [comment]
Sending #620 for the remaining issue, essentially fixing the known issues introduced by this PR.

**Pull request merged**